plugins/chat/app/models/chat/channel.rb

```diff
@@ -107,6 +107,19 @@ def self.ensure_consistency!
       update_counts
     end
 
+    def can_be_joined_by?(user)
+      guardian = Guardian.new(user)
```

Comment:
User is supposed to have a guardian: `user.guardian` (which is a shorthand for what you do here)